### ISSUE TITLE
BitBucket Updates not being detected (anymore)

### ISSUE THREAD

**Issue by hirasso** [YahnisElsts/plugin-update-checker#317]
Hi there, 

I am hosting all my custom plugins on my BitBucket Account and am using your great script for updates detection. But yesterday I noticed that updates are not being detected anymore. No complaints from the script, just no update notice in my WP plugins list... Is there any way I could retrieve information to debug this? Like a debug mode?

I checked for updates for one of my plugins using debug bar:

```
Puc_v4p6_Plugin_Info Object
(
    [name] => RH WPSC Clear Cache
    [slug] => rh-wpsc-clear-cache
    [version] => 
    [homepage] => 
    [sections] => Array
        (
            [description] => Replaces the default 'Delete Cache' Button from WP Admin menu and makes it also available for editors
            [changelog] => There is no changelog available.
        )

    [download_url] => https://bitbucket.org/rhilber/rh-wpsc-clear-cache/get/master.zip
    [banners] => 
    [icons] => Array
        (
        )

    [translations] => Array
        (
        )

    [author] => Rasso Hilber
    [author_homepage] => https://rassohilber.com
    [requires] => 
    [tested] => 
    [upgrade_notice] => 
    [rating] => 
    [num_ratings] => 
    [downloaded] => 
    [active_installs] => 
    [last_updated] => 2019-09-23T16:35:17+00:00
    [id] => 0
    [filename] => rh-wpsc-clear-cache/rh-wpsc-clear-cache.php
)
```

...oh, and just found the status checker button in debug bar. This is it's response:

```
No updates found.
The update checker encountered 2 API errors.

1) BitBucket API error. Base URL: "https://api.bitbucket.org/1.0/repositories/rhilber/rh-wpsc-clear-cache/src/master/readme.txt", HTTP status code: 410.
Error code:
puc-bitbucket-http-error
Requested URL:
https://api.bitbucket.org/1.0/repositories/rhilber/rh-wpsc-clear-cache/src/master/readme.txt?oauth_consumer_key=T764zhLfbchzpCfq8m&oauth_nonce=c6eb6a6c9f5fa13ba5e9f524be82f600&oauth_signature_method=HMAC-SHA1&oauth_timestamp=1569495263&oauth_version=1.0&oauth_signature=CDy7gKxe%2BogOh0plPwSnmDezZ1M%3D
HTTP status:
410 Gone
Response headers:
vary: User-Agent
content-type: application/json; charset=utf-8
date: Thu, 26 Sep 2019 10:54:23 GMT
content-length: 262
Response body:
{"type": "error", "error": {"message": "Resource removed", "detail": "This API is no longer supported.\n\nFor information about its removal, please refer to the deprecation notice at: https://developer.atlassian.com/cloud/bitbucket/deprecation-notice-v1-apis/"}}
2) BitBucket API error. Base URL: "https://api.bitbucket.org/1.0/repositories/rhilber/rh-wpsc-clear-cache/src/master/rh-wpsc-clear-cache.php", HTTP status code: 410.
Error code:
puc-bitbucket-http-error
Requested URL:
https://api.bitbucket.org/1.0/repositories/rhilber/rh-wpsc-clear-cache/src/master/rh-wpsc-clear-cache.php?oauth_consumer_key=T764zhLfbchzpCfq8m&oauth_nonce=09fff0abd43b099eba411b72567bb5d2&oauth_signature_method=HMAC-SHA1&oauth_timestamp=1569495265&oauth_version=1.0&oauth_signature=cWv2cZbRIhcLiYhzUds%2FshBEkIs%3D
HTTP status:
410 Gone
Response headers:
vary: User-Agent
content-type: application/json; charset=utf-8
date: Thu, 26 Sep 2019 10:54:25 GMT
content-length: 262
Response body:
{"type": "error", "error": {"message": "Resource removed", "detail": "This API is no longer supported.\n\nFor information about its removal, please refer to the deprecation notice at: https://developer.atlassian.com/cloud/bitbucket/deprecation-notice-v1-apis/"}}
```

Maybe it has something to do with the latest bitbucket update?

**Comment by hirasso** [comment]
So if I go to this endpoint directly:

[https://api.bitbucket.org/1.0/repositories/rhilber/rh-wpsc-clear-cache/src/master/rh-wpsc-clear-cache.php](https://api.bitbucket.org/1.0/repositories/rhilber/rh-wpsc-clear-cache/src/master/rh-wpsc-clear-cache.php)

I get this response:

```json
{
type: "error",
error: {
message: "Resource removed",
detail: "This API is no longer supported. For information about its removal, please refer to the deprecation notice at: https://developer.atlassian.com/cloud/bitbucket/deprecation-notice-v1-apis/"
}
}
```
So it seems to be related to a bitbucket api change, indeed. Maybe this helps you in fixing it?

**Comment by YahnisElsts** [comment]
It looks like you might be using an older version of plugin-update-checker. Try updating to version 4.7. Older versions may no longer work properly due to BitBucket API changes.

**Comment by hirasso** [comment]
Thanks @YahnisElsts , that was definitely it. Sorry for bothering you. Closing this.
